### PR TITLE
Allow using build definition name in 'Build definition' input

### PR DESCRIPTION
**Pull request by lypolyzo**
Here is an example of usage:
image
In this case the definitionId would resolve in a name rather than a number

### CODE REVIEW

File: Tasks/DownloadBuildArtifacts/task.loc.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "author": "Microsoft Corporation",
   "version": {
     "Major": 0,
-    "Minor": 132,
+    "Minor": 134,
```
Comment (lypolyzo) [inline Tasks/DownloadBuildArtifacts/task.loc.json]: These files where automatically updated. I'm not sure if this an expected behavior

File: Tasks/DownloadBuildArtifacts/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -12,6 +12,9 @@
   "loc.input.help.definition": "Select the build definition name",
   "loc.input.label.specificBuildWithTriggering": "When appropriate, download artifacts from the triggering build.",
   "loc.input.help.specificBuildWithTriggering": "If checked, this build task will try to download artifacts from the triggering build. If there is no triggering build from the specified definition, it will download artifacts from the build specified in the options below.",
+  "loc.input.label.buildVersionToDownload": "Build version to download",
```
Comment (lypolyzo) [inline Tasks/DownloadBuildArtifacts/Strings/resources.resjson/en-US/resources.resjson]: This file was automatically updated. Not sure if this is the expected behavior

File: Tasks/DownloadBuildArtifacts/main.ts [hunk 3]
```diff
@@ -133,6 +133,17 @@ async function main(): Promise<void> {
             }
         }
 
+        // if the definition name includes a variable then definitionIdSpecified is a name vs a number
+        if (Number.isNaN(parseInt(definitionIdSpecified))) {
+            var definitions: BuildDefinitionReference[] = await executeWithRetries("getBuildDefinitions", () => buildApi.getDefinitions(projectId, definitionIdSpecified), 4).catch((reason) => {
+                reject(reason);
+                return;
+            });
+            definitionId = String(definitions[0].id);
```
Comment (omeshp) [inline Tasks/DownloadBuildArtifacts/main.ts]: check if definitions array isn't empty/undefined/null

**Comment by stephenmichaelf** [comment]
@lypolyzo Could you follow @GitHubSriramB 's suggestion? Thanks!

File: Tasks/DownloadBuildArtifacts/main.ts (hunk 3 above)
Comment (lypolyzo) [reply Tasks/DownloadBuildArtifacts/main.ts]: I will update the if statement in line 148 (previous 137) instead. Found [here](https://stackoverflow.com/questions/28975896/is-there-a-dedicated-function-to-check-null-and-undefined-in-typescript) the if statement checks for all of the above

**Comment by lypolyzo** [comment]
Using [7879](https://github.com/Microsoft/vsts-tasks/pull/7879) to track this change

**Pull request closed without merging**